library/std/src/time.rs

```diff
@@ -125,6 +126,25 @@ pub use core::time::FromSecsError;
 /// > structure cannot represent the new point in time.
 ///
 /// [`add`]: Instant::add
+///
+/// ## Monotonicity
+///
+/// All [tier 1] and most tier 2 platforms with std support rely on monotonicity guarantees
+/// from the operating system. Rarely these guarantees are broken by hardware, virtualization
+/// or operating system bugs. To work around these bugs and platforms not offering monotonic clocks
+/// [`duration_since`], [`elapsed`] and [`sub`] saturate to zero. In older rust versions this
+/// lead to a panic instead. [`checked_duration_since`] can be used to detect and handle situations
+/// where monotonicity is violated.
```

Comment:
```suggestion
/// where monotonicity is violated, or `Instant`s are subtracted in the wrong order.
```

Not quite sure about this wording, but I feel like the previous language made checked_duration_since sound like a magic wand a little more than we probably want it to; 99% of users likely won't want to use it.

Can we add some language in this PR to checked_duration_since that points to this paragraph, and says something to the effect of "due to the bugs discussed, even under correct logical ordering of the passed instants, this method can return None".